src/App.js

```diff
@@ -7,15 +7,24 @@ import Home from "./pages/home/Home";
 import Search from "./pages/search/Search";
 import Login from "./pages/login/Login";
 
+import rootStore from "./store/root-store";
+
 function App() {
   const location = useLocation();
   const shouldShowHeaderAndSidenav = location.pathname !== "/login";
   const [searchParams] = useSearchParams();
 
+  useEffect(() => {
+    if (localStorage.getItem("token")) {
+      rootStore.userStore.setIsLoggedIn(true);
+    }
+  }, []);
```

Comment:
if you recall, useEffect is a hook that react provides where we can tell our application to do anything we want in the react component life cycles

There are three important life cycles you should be aware of:
1. When a component mounts (when the component is visible on the screen)
2. When a component has been updated (when the _state_ of the component is updated for example if the user is logged in or not)
3. When a component unmounts (the component is no longer visible or doesn't exist on the screen)

In the code above, we're pretty much telling `App.jsx` to make sure that we set the `loggedIn` variable in the user store to true if the token exists in local storage. The `loggedIn` variable will then be used to show a different _state_ of the application when the user is logged in